frame/support/procedural/src/pallet/parse/storage.rs

```diff
@@ -514,10 +514,16 @@ fn process_unnamed_generics(
 	})?;
 
 	let use_default_hasher = |arg_pos| {
-		if let Some(arg) = retrieve_arg(arg_pos).ok() {
-			dev_mode && syn::parse2::<syn::Token![_]>(arg.to_token_stream()).is_ok()
+		let arg = retrieve_arg(arg_pos)?;
+		if syn::parse2::<syn::Token![_]>(arg.to_token_stream()).is_ok() {
+			if dev_mode {
+				Ok(true)
+			} else {
+				let msg = "`_` can only be used in dev_mode. Please specify an appropriate hasher.";
+				Err(syn::Error::new(args_span, msg))
```

Comment:
```suggestion
				Err(syn::Error::new(arg.span(), msg))
```

To highlight exactly which argument we mean.

Reply:
Done. Thank you.